### PR TITLE
Add maximum number of timesteps to applications

### PR DESCRIPTION
**Pull request by sebglane**
Changes:

- terminal printing
- remove unnecessary method in application `step-35.cc`
- make all applications run 10 time steps by default
- reduce refinement in `Christensen.cc`

### CODE REVIEW

**Comment by sebglane** [comment]
Merge after #122

File: applications/Christensen.cc [hunk 1]
```diff
@@ -460,7 +454,10 @@ void Christensen<dim>::run()
   temperature->solution = temperature->old_solution;
   output();
 
-  while (time_stepping.get_current_time() < time_stepping.get_end_time())
+  const unsigned int n_steps = this->prm.time_discretization_parameters.n_maximum_steps;
+
+  while (time_stepping.get_current_time() < time_stepping.get_end_time() &&
+         time_stepping.get_step_number() < n_steps)
```
Comment (j507) [inline applications/Christensen.cc:460]: Could a bypass case be introduced? For example, when `.n_maximum_steps == 0` it is ignored. This avoids having to compute the number of steps based upon the time step and end time values for each benchmark.

**Review by j507 — APPROVED** [review]
A bypass should be implemented in my opinion. Nonetheless, feel free to merge.

File: applications/Christensen.cc (hunk 1 above)
Comment (sebglane) [reply applications/Christensen.cc]: I implemented this in 9cb0306ced3c70b7d825aa858a8dc03f0e8c9924

**Pull request merged**